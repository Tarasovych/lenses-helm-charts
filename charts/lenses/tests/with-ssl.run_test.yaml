suite: enable SSL and check the changes in deployment, configmap and secret
templates:
  - deployment.yaml
  - configmap.yaml
  - secrets.yaml
tests:
  - it: should work
    values:
      - values/with-ssl.yaml
    asserts:
      # Check the rendered manifest has not changed
      # Review the changes and if they are verified use `helm unittest -u .` to update the manifest
      # Writing tests that cover this chaneg is highly recommended
      - matchSnapshot: {}
      # deployment.yaml
      - template: deployment.yaml
        isKind:
          of: Deployment
      - template: deployment.yaml
        contains:
          path: spec.template.spec.containers[0].env
          content:
            name: LENSES_KAFKA_BROKERS
            value: SSL://kafka:9093
      - template: deployment.yaml
        contains:
          path: spec.template.spec.containers[0].env
          content:
            name: LENSES_KAFKA_SETTINGS_CLIENT_SECURITY_PROTOCOL
            value: SSL
      - template: deployment.yaml
        contains:
          path: spec.template.spec.containers[0].env
          content:
            name: LENSES_KAFKA_SETTINGS_CLIENT_SSL_TRUSTSTORE_LOCATION
            value: /mnt/secrets/client.truststore.jks
      - template: deployment.yaml
        contains:
          path: spec.template.spec.containers[0].env
          content:
            name: LENSES_KAFKA_SETTINGS_CLIENT_SSL_KEYSTORE_LOCATION
            value: /mnt/secrets/client.keystore.jks
      - template: deployment.yaml
        contains:
          path: spec.template.spec.containers[0].env
          content:
            name: LENSES_KAFKA_SETTINGS_CLIENT_SSL_TRUSTSTORE_PASSWORD
            valueFrom:
              secretKeyRef:
                key: client.truststore.password
                name: RELEASE-NAME
      - template: deployment.yaml
        contains:
          path: spec.template.spec.containers[0].env
          content:
            name: LENSES_KAFKA_SETTINGS_CLIENT_SSL_KEYSTORE_PASSWORD
            valueFrom:
              secretKeyRef:
                key: client.keystore.password
                name: RELEASE-NAME
      - template: deployment.yaml
        contains:
          path: spec.template.spec.containers[0].env
          content:
            name: LENSES_KAFKA_SETTINGS_CLIENT_SSL_KEY_PASSWORD
            valueFrom:
              secretKeyRef:
                key: client.key.password
                name: RELEASE-NAME
      # configmap.yaml
      - template: configmap.yaml
        matchSnapshot: {}
      - template: configmap.yaml
        isKind:
          of: ConfigMap

      # secrets.yaml
      - template: secrets.yaml
        matchSnapshot: {}
      - template: secrets.yaml
        isKind:
          of: Secret
