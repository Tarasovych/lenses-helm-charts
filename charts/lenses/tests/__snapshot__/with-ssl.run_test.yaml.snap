should work:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations: null
      labels:
        app: RELEASE-NAME
        chart: lenses-4.2.13
        component: lenses
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses
        release: RELEASE-NAME
      name: RELEASE-NAME
    spec:
      replicas: 1
      selector:
        matchLabels:
          app: RELEASE-NAME
      strategy:
        type: Recreate
      template:
        metadata:
          annotations:
            prometheus.io/path: /metrics
            prometheus.io/port: "9102"
            prometheus.io/scrape: "true"
          labels:
            app: RELEASE-NAME
            component: lenses
            containerPort: "3030"
            hostPort: "3030"
        spec:
          containers:
          - env:
            - name: LENSES_OPTS
              value: ""
            - name: LENSES_PORT
              value: "3030"
            - name: LENSES_JMX_PORT
              value: "9101"
            - name: LENSES_KAFKA_BROKERS
              value: SSL://kafka:9093
            - name: LENSES_SCHEMA_REGISTRY_URLS
              value: ""
            - name: LENSES_SCHEMA_REGISTRY_MODE
              value: ""
            - name: LENSES_KAFKA_CONNECT_CLUSTERS
              valueFrom:
                secretKeyRef:
                  key: connect.clusters
                  name: RELEASE-NAME
            - name: LENSES_ALERT_CONSUMERS_LAG_THRESHOLD
              value: "50000"
            - name: LENSES_ALERT_PLUGINS
              value: ""
            - name: LENSES_TOPICS_AUDITS
              value: _kafka_lenses_audits
            - name: LENSES_TOPICS_METRICS
              value: _kafka_lenses_metrics
            - name: LENSES_TOPICS_PROFILES
              value: _kafka_lenses_profiles
            - name: LENSES_TOPICS_PROCESSORS
              value: _kafka_lenses_processors
            - name: LENSES_TOPICS_CLUSTER
              value: _kafka_lenses_cluster
            - name: LENSES_TOPICS_ALERTS_STORAGE
              value: _kafka_lenses_alerts
            - name: LENSES_TOPICS_ALERTS_SETTINGS
              value: _kafka_lenses_alert_settings
            - name: LENSES_TOPICS_LSQL_STORAGE
              value: _kafka_lenses_lsql_storage
            - name: LENSES_TOPICS_METADATA
              value: _kafka_lenses_topics_metadata
            - name: LENSES_TOPICS_EXTERNAL_TOPOLOGY
              value: __topology
            - name: LENSES_SCHEMA_REGISTRY_TOPICS
              value: null
            - name: LENSES_SQL_EXECUTION_MODE
              value: IN_PROC
            - name: LENSES_SECRET_FILE
              value: /mnt/secrets/security.conf
            - name: LENSES_SCHEMA_REGISTRY_AUTH
              value: ""
            - name: LENSES_KAFKA_SETTINGS_CLIENT_BASIC_AUTH_CREDENTIALS_SOURCE
              value: ""
            - name: LENSES_KAFKA_SETTINGS_CLIENT_SECURITY_PROTOCOL
              value: SSL
            - name: LENSES_KAFKA_SETTINGS_CLIENT_SASL_MECHANISM
              value: GSSAPI
            - name: LENSES_KAFKA_SETTINGS_CLIENT_SSL_TRUSTSTORE_LOCATION
              value: /mnt/secrets/client.truststore.jks
            - name: LENSES_KAFKA_SETTINGS_CLIENT_SSL_KEYSTORE_LOCATION
              value: /mnt/secrets/client.keystore.jks
            - name: LENSES_KAFKA_SETTINGS_CLIENT_SSL_TRUSTSTORE_PASSWORD
              valueFrom:
                secretKeyRef:
                  key: client.truststore.password
                  name: RELEASE-NAME
            - name: LENSES_KAFKA_SETTINGS_CLIENT_SSL_KEYSTORE_PASSWORD
              valueFrom:
                secretKeyRef:
                  key: client.keystore.password
                  name: RELEASE-NAME
            - name: LENSES_KAFKA_SETTINGS_CLIENT_SSL_KEY_PASSWORD
              valueFrom:
                secretKeyRef:
                  key: client.key.password
                  name: RELEASE-NAME
            - name: LENSES_DEPLOYMENTS_EVENTS_BUFFER_SIZE
              value: "10000"
            - name: LENSES_DEPLOYMENTS_ERRORS_BUFFER_SIZE
              value: "1000"
            - name: LENSES_DEPLOYMENT_CONNECT_STATUS_INTERVAL
              value: 30 second
            - name: LENSES_DEPLOYMENT_CONNECT_ACTIONS_BUFFER_SIZE
              value: "1000"
            - name: LC_KUBERNETES_MODE
              value: "true"
            - name: LT_PACKAGE
              value: helm
            - name: LT_PACKAGE_VERSION
              value: 4.2.13
            image: lensesio/lenses:4.2.6
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /
                port: 3030
              initialDelaySeconds: 300
              periodSeconds: 10
            name: lenses
            ports:
            - containerPort: 3030
            resources:
              limits:
                memory: 5Gi
              requests:
                memory: 4Gi
            volumeMounts:
            - mountPath: /mnt/secrets
              name: secrets
            - mountPath: /etc/keytab
              name: secrets
              subPath: keytab
            - mountPath: /etc/registry.keytab
              name: secrets
              subPath: registry.keytab
            - mountPath: /mnt/settings/security.append.conf
              name: secrets
              subPath: security.append.conf
            - mountPath: /mnt/settings/lenses.append.conf
              name: config
              subPath: lenses.append.conf
            - mountPath: /data/storage
              name: storage
          serviceAccountName: default
          volumes:
          - name: secrets
            secret:
              items:
              - key: security.conf
                path: security.conf
              - key: security.append.conf
                path: security.append.conf
              - key: lenses.keytab
                path: lenses.keytab
              - key: keytab
                path: keytab
              - key: client.keystore.jks
                path: client.keystore.jks
              - key: client.truststore.jks
                path: client.truststore.jks
              - key: jaas.conf
                path: jaas.conf
              - key: schema.registry.keystore.jks
                path: schema.registry.keystore.jks
              - key: schema.registry.truststore.jks
                path: schema.registry.truststore.jks
              - key: lenses.opts.keystore.jks
                path: lenses.opts.keystore.jks
              - key: lenses.opts.truststore.jks
                path: lenses.opts.truststore.jks
              secretName: RELEASE-NAME
          - configMap:
              name: RELEASE-NAME
            name: config
          - name: storage
            persistentVolumeClaim:
              claimName: RELEASE-NAME-claim
  2: |
    apiVersion: v1
    data:
      lenses.append.conf: ""
    kind: ConfigMap
    metadata:
      labels:
        app: RELEASE-NAME
        chart: lenses-4.2.13
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses-configmap
        release: RELEASE-NAME
      name: RELEASE-NAME
  3: |
    apiVersion: v1
    data:
      client.keystore.jks: bar
      client.truststore.jks: foo
      keytab: ""
      lenses.keytab: ""
      lenses.opts.keystore.jks: ""
      lenses.opts.truststore.jks: ""
      lenses.tls.keystore.jks: ""
      lenses.tls.truststore.jks: ""
      saml.idp.xml: ""
      saml.keystore.jks: ""
    kind: Secret
    metadata:
      labels:
        app: RELEASE-NAME
        chart: lenses-4.2.13
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses-secret
        release: RELEASE-NAME
      name: RELEASE-NAME
    stringData:
      client.key.password: foobarPass
      client.keystore.password: barPass
      client.truststore.password: fooPass
      connect.clusters: ""
      jaas.conf: ""
      jvm.truststore.jks: ""
      jvm.truststore.password: ""
      lenses.opts.keystore.password: ""
      lenses.opts.truststore.password: ""
      lenses.tls.key.password: ""
      lenses.tls.keystore.password: ""
      lenses.tls.truststore.password: ""
      license: ""
      schema.registry.key.password: ""
      schema.registry.keystore.jks: ""
      schema.registry.keystore.password: ""
      schema.registry.truststore.jks: ""
      schema.registry.truststore.password: ""
      security.append.conf: ""
      security.conf: ""
  4: |
    apiVersion: v1
    data:
      lenses.append.conf: ""
    kind: ConfigMap
    metadata:
      labels:
        app: RELEASE-NAME
        chart: lenses-4.2.13
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses-configmap
        release: RELEASE-NAME
      name: RELEASE-NAME
  5: |
    apiVersion: v1
    data:
      client.keystore.jks: bar
      client.truststore.jks: foo
      keytab: ""
      lenses.keytab: ""
      lenses.opts.keystore.jks: ""
      lenses.opts.truststore.jks: ""
      lenses.tls.keystore.jks: ""
      lenses.tls.truststore.jks: ""
      saml.idp.xml: ""
      saml.keystore.jks: ""
    kind: Secret
    metadata:
      labels:
        app: RELEASE-NAME
        chart: lenses-4.2.13
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses-secret
        release: RELEASE-NAME
      name: RELEASE-NAME
    stringData:
      client.key.password: foobarPass
      client.keystore.password: barPass
      client.truststore.password: fooPass
      connect.clusters: ""
      jaas.conf: ""
      jvm.truststore.jks: ""
      jvm.truststore.password: ""
      lenses.opts.keystore.password: ""
      lenses.opts.truststore.password: ""
      lenses.tls.key.password: ""
      lenses.tls.keystore.password: ""
      lenses.tls.truststore.password: ""
      license: ""
      schema.registry.key.password: ""
      schema.registry.keystore.jks: ""
      schema.registry.keystore.password: ""
      schema.registry.truststore.jks: ""
      schema.registry.truststore.password: ""
      security.append.conf: ""
      security.conf: ""
